hare/broker.go

```diff
@@ -308,9 +309,11 @@ func (b *Broker) Register(id instanceID) (chan *Msg, error) {
 		b.updateLatestLayer(id)
 
 		if len(b.outbox) >= b.limit {
-			resErr <- errTooMany
-			resCh <- nil
-			return
+			//unregister the earliest layer to make space for the new layer
+			//cannot call unregister here because unregister blocks and this would cause a deadlock
+			instance := b.minDeleted + 1
```

Comment:
I think you only want to unregister the oldest layer after the `b.isSynced(id)` check, below